plugins/auth-backend/src/providers/oidc/provider.ts

```diff
@@ -14,269 +14,56 @@
-
-  private async setupStrategy(options: Options): Promise<OidcImpl> {
-    const issuer = await Issuer.discover(options.metadataUrl);
-    const client = new issuer.Client({
-      access_type: 'offline', // this option must be passed to provider to receive a refresh token
-      client_id: options.clientId,
-      client_secret: options.clientSecret,
-      redirect_uris: [options.callbackUrl],
-      response_types: ['code'],
-      token_endpoint_auth_method:
-        options.tokenEndpointAuthMethod || 'client_secret_basic',
-      id_token_signed_response_alg: options.tokenSignedResponseAlg || 'RS256',
-      scope: options.scope || '',
-    });
-
-    const strategy = new OidcStrategy(
-      {
-        client,
-        passReqToCallback: false,
-      },
-      (
-        tokenset: TokenSet,
-        userinfo: UserinfoResponse,
-        done: PassportDoneCallback<OidcAuthResult, PrivateInfo>,
-      ) => {
-        if (typeof done !== 'function') {
-          throw new Error(
-            'OIDC IdP must provide a userinfo_endpoint in the metadata response',
-          );
-        }
-        done(
-          undefined,
-          { tokenset, userinfo },
-          {
-            refreshToken: tokenset.refresh_token,
-          },
-        );
-      },
-    );
-    strategy.error = console.error;
-    return { strategy, client };
-  }
-
-  // Use this function to grab the user profile info from the token
-  // Then populate the profile with it
-  private async handleResult(result: OidcAuthResult): Promise<OAuthResponse> {
-    const { profile } = await this.authHandler(result, this.resolverContext);
-
-    const expiresInSeconds =
-      result.tokenset.expires_in === undefined
-        ? BACKSTAGE_SESSION_EXPIRATION
-        : Math.min(result.tokenset.expires_in, BACKSTAGE_SESSION_EXPIRATION);
-
-    let backstageIdentity = undefined;
-    if (this.signInResolver) {
-      backstageIdentity = await this.signInResolver(
-        {
-          result,
-          profile,
-        },
-        this.resolverContext,
-      );
-    }
-
-    return {
-      backstageIdentity,
-      providerInfo: {
-        idToken: result.tokenset.id_token,
-        accessToken: result.tokenset.access_token!,
-        scope: result.tokenset.scope!,
-        expiresInSeconds,
-      },
-      profile,
-    };
-  }
-}
-
 /**
  * Auth provider integration for generic OpenID Connect auth
  *
  * @public
  */
 export const oidc = createAuthProviderIntegration({
   create(options?: {
-    authHandler?: AuthHandler<OidcAuthResult>;
+    /**
+     * The profile transformation function used to verify and convert the auth response
+     * into the profile that will be presented to the user.
+     */
+    authHandler?: AuthHandler<OAuthResult>;
 
+    /**
+     * Configure sign-in for this provider, without it the provider can not be used to sign users in.
+     */
     signIn?: {
-      resolver: SignInResolver<OidcAuthResult>;
+      resolver: SignInResolver<OAuthResult>;
```

Comment:
These type changes are actually breaking because the `OidcAuthResult` and `OAuthResult` types are incompatible: `OAuthResult` has fields like `fullProfile` and `params`, whereas `OidcAuthResult` has `tokenset` and `userinfo`.

Reply:
reverted in fd32e0d